src/plugins/mer/meremulatormodedialog.cpp

```diff
@@ -243,3 +232,9 @@ void MerEmulatorModeDialog::guessOptimalViewMode()
         : m_ui->originalViewModeRadioButton;
     viewModeRadioButton->setChecked(true);
 }
+
+void MerEmulatorModeDialog::onDeviceListReplaced()
+{
+    if (m_kit != nullptr)
+        onKitUpdated(m_kit);
+}
```

Comment:
This requires the reader to locate onKitUpdated in order to understand what really happens here - just to save two lines of code. IMO it would be better to define the slot next to onKitUpdated and say clearly

    if (m_kit == nullptr)
        return;

    auto device = DeviceKitInformation::device(m_kit);
    setEmulator(device.dynamicCast<const MerEmulatorDevice>().constCast<MerEmulatorDevice>());

Reply:
Changed as you said. PS I pondered how to better to do this and choose less code instead of readability.

Reply:
Yeah I was a bit unsure too, but we know, code is written once, read many times :) Thanks